astropy/table/table.py

```diff
@@ -3643,7 +3644,10 @@ def _convert_col_for_table(self, col):
             # We need to turn the column into a quantity, or a subclass
             # identified in the unit (such as u.mag()).
             q_cls = getattr(col.unit, '_quantity_class', Quantity)
-            qcol = q_cls(col.data, col.unit, copy=False)
```

Comment:
I would change this to `Unit(col.unit)` and omit the try/except.

Reply:
Unit(col.unit) does not fix the problem, the intermediate conversion to a string is necessary.

If Unit() cannot parse the unit then it will throw a ValueError. The try/except statement ensures that if this happens the code will revert to the old behavior. Omitting try/except could make some files unreadable.